core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -184,11 +184,23 @@ private[spark] class TaskSetManager(
     t.epoch = epoch
   }
 
+  // An array to store preferred location and its task index
+  private val locationWithTaskIndex: ArrayBuffer[(String, Int)] = new ArrayBuffer[(String, Int)]()
```

Comment:
All these variables are now stored as fields in this class, and the only one you reference outside of this code is  `locationWithTaskIndex`.  This should be reworked to avoid that.

Reply:
Fixed.